src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelEventSource.cs

```diff
@@ -215,6 +219,54 @@ public void TlsHandshakeFailed(string connectionId)
             }
         }
 
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        [Event(11, Level = EventLevel.Informational)]
+        public void Configuration(int instanceId, string configuration)
+        {
+            // If the event source is already enabled, dump configuration
+            WriteEvent(11, instanceId, configuration);    
+        }
+
+        [NonEvent]
+        public void Configuration(KestrelServerOptions options)
+        {
+            var bufferWriter = new ArrayBufferWriter<byte>();
```

Comment:
Is there an equivalent of checking log level here? Only want to write JSON if Informational and above events are being used.

Reply:
This:

https://github.com/dotnet/aspnetcore/blob/e470d40c808ca505c8e88bd4678315e6b5cf07b1/src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelEventSource.cs#L63

Reply:
This is checked in AddServerOptions.

Reply:
The Configuration event is invoked from OnEventCommand and that path ideally would also test the level. If that level test was here it would make it easy to have it enforced uniformly. Currently Configuration() is also a public API so any other code could become a new caller in the future.